### ISSUE TITLE
Flaky workspace with no outputs test

### ISSUE THREAD

**Issue by yitsushi** [weaveworks/tf-controller#719]
Sometimes this test fails but it shouldn't:
```
GIVEN: a Terraform resource with auto approve, attached to the given GitRepository resource.
BY: creating a new TF resource and attaching to the repo via `sourceRef`.
IT should be created and attached successfully.
    tc000011_workspace_no_outputs_test.go:101: 
        Expected success, but got an error:
            <*errors.StatusError | 0xc0004690e0>: 
            object is being deleted: terraforms.infra.contrib.fluxcd.io "helloworld-no-outputs" already exists
            {
                ErrStatus: {
                    TypeMeta: {Kind: "", APIVersion: ""},
                    ListMeta: {
                        SelfLink: "",
                        ResourceVersion: "",
                        Continue: "",
                        RemainingItemCount: nil,
                    },
                    Status: "Failure",
                    Message: "object is being deleted: terraforms.infra.contrib.fluxcd.io \"helloworld-no-outputs\" already exists",
                    Reason: "AlreadyExists",
                    Details: {
                        Name: "helloworld-no-outputs",
                        Group: "infra.contrib.fluxcd.io",
                        Kind: "terraforms",
                        UID: "",
                        Causes: nil,
                        RetryAfterSeconds: 0,
                    },
                    Code: 409,
                },
            }
--- FAIL: Test_000011_workspace_no_outputs_test (0.05s)
=== RUN   Test_000012_src_bucket_no_outputs_test
```

Example:
https://github.com/weaveworks/tf-controller/actions/runs/5400939142/jobs/9810067445?pr=715

========

User Story:

As a Terraform controller developer,
I'd like to address the flaky test "workspace with no outputs"
so that the test runs consistently and does not fail due to race conditions or other unnecessary conflicts.

Acceptance Criteria:
- [ ] The "workspace with no outputs" test should not fail intermittently when the Terraform resource and GitRepository resource are created and attached properly.
- [ ] The error message "object is being deleted: terraforms.infra.contrib.fluxcd.io \"helloworld-no-outputs\" already exists" should be resolved.
- [ ] The test should pass consistently on different CI runs.

**Comment by yiannistri** [comment]
This will be removed/fixed as part of https://github.com/weaveworks/tf-controller/issues/856

**Comment by yiannistri** [comment]
Closing as it is addressed in #856
